File: CSC205-Java/_solutions/Module02-Inheritance-and-Object-Oriented-Design/debug02/BankAccount.java
// A basic bank account with deposits and withdrawals.
public class BankAccount {
    private String owner;
    protected double balance;

    public BankAccount(String owner, double openingBalance) {
        this.owner = owner;
        this.balance = openingBalance;
    }

    public void deposit(double amount) {
        if (amount > 0) {
            balance += amount;
        }
    }

    public boolean withdraw(double amount) {
        if (amount <= balance) {
            balance -= amount;
            return true;
        }
        return false;
    }

    public double getBalance() {
        return balance;
    }

    public String getOwner() {
        return owner;
    }
}

File: CSC101-Python/Module11-Classes-and-Objects/debug01/bank_account.py
# bank_account.py
# A simple bank account that supports deposits and withdrawals.


class BankAccount:
    def __init__(self, owner, balance=0.0):
        self.owner = owner
        self.balance = balance

    def deposit(self, amount):
        balance = self.balance + amount
        print(f"Deposited ${amount:.2f}")

    def withdraw(self, amount):
        if amount > self.balance:
            print("Insufficient funds.")
        else:
            self.balance = self.balance - amount
            print(f"Withdrew ${amount:.2f}")

    def __str__(self):
        return f"{self.owner}: ${self.balance:.2f}"


account = BankAccount("Jordan", 50.00)
print(account)
account.deposit(25.00)
account.withdraw(60.00)
print(account)
